documentation/mqtt-overview.md

```diff
@@ -0,0 +1,88 @@
+# AnkerMake MQTT Communications Overview
+
+We're going over the the [MQTT communications standard](https://mqtt.org/) used by AnkerMake. While we won't be explaining the MQTT standard itself in much detail, we will explain certain aspects of how the protocol works for ease of understanding.
+
+## Generic MQTT Basics
+
+MQTT (Messaging Queue Transport Telemetry) is a lightweight and low bandwidth communications protocol primarily used by [IoT devices](https://en.wikipedia.org/wiki/Internet_of_things).
+
+The two actors in this type of network are a **Broker** and a **Client**.  MQTT uses a publish/subscribe model to pass data and commands between clients facilitated by a broker. In order to organize where data and commands are supposed to go, the Broker organizes a hierarchy of **Topics**. 
+
+A Topic is simply a namespace that clients can subscribe and/or publish to. 
+
+> For example, consider modern chat/messaging applications such as iMessage and Discord. Most of these applications have implemented a feature that shows all the participants of a chat that another participant is typing a response. The topic for this feature may look something like `chat-app/conversation/participant-a/is-typing` and have a boolean value set to "false" by default. 
+>
+> When Participant A starts to type in the compose field, their client would publish an update to this topic that sets the value to "true". Each participant would have a corresponding topic that represents that they are typing and the other participant's clients would be subscribed to these topics. When the value of these topics changes to "true", each participant's client would present an animation to communicate this to the user.
+
+The Broker is responsible for establishing client-initiated connections and forwarding messages sent by the originating client to other subscribed clients.
```

Comment:
Correct me if I'm misunderstanding, but my reasoning goes like this: For a TLS connection to form, both sides need to handshake. What I'm trying to say here is that the hanshake is intiated by a client and then agreed upon by the broker. The broker never reaches out to a client first, because it has no idea they exist prior to them attempting a connection.

I guess that is kinda obvious and hence why you're asking. How can I re-phrase that to either make it more clear or should I just remove the unecessary info?

Reply:
Now I see :)

I think we can make it simpler for everybody (including ourselves), by saying something like:

"MQTT messsages are sent over a transport connection between the client and the broker. This connection can be unencrypted or use TLS encryption, and will always be initiated from the client side."